lib/src/barback.dart

```diff
@@ -87,3 +89,43 @@ AssetId packagesUrlToId(Uri url) {
   var assetPath = p.url.join("lib", p.url.joinAll(parts.skip(index + 2)));
   return new AssetId(package, assetPath);
 }
+
+/// Convert [importUri] found in [source] to an [AssetId], handling both
+/// `package:` imports and relative imports.
+///
+/// Returns [null] for `dart:` uris since they cannot be referenced properly
+///  by an `AssetId`.
+///
+/// Throws an [ArgumentError] if an [AssetId] can otherwise not be created.
+AssetId importUriToAssetId(AssetId source, String importUri) {
+  var parsedUri = Uri.parse(importUri);
+  if (parsedUri.isAbsolute) {
+    if (parsedUri.scheme == 'package') {
```

Comment:
done, also added a better comment highlighting the cases in which this method will throw